### PR TITLE
[DEV-11466] Fix MarkSubmissionAsRetrieved

### PR DESCRIPTION
**Pull request by nateshim-indico**
- Fix MarkSubmissionRetrieved Lag by using GetAsync instead of ListAsync
- Update dotnet version to 6.0
- Update GetSubmission call to match ListSubmissions in returning full Submission data from GraphQL call

### CODE REVIEW

**Comment by goatrocks** [comment]
needs a JIRA ticket please 🙏

File: IndicoV2/Submissions/SubmissionsClient.cs [hunk 1]
```diff
@@ -145,12 +145,22 @@ public async Task<ISubmission> GetAsync(int submissionId, CancellationToken canc
 
         public async Task<ISubmission> MarkSubmissionAsRetrieved(int submissionId, bool retrieved = true, CancellationToken cancellationToken = default)
         {
-            var resultId = await _strawberryShakeClient.Submissions().MarkRetrieved(submissionId, retrieved, cancellationToken);
-            var result = await _strawberryShakeClient.Submissions().List(new List<int?>(submissionId).AsReadOnly(), default, default, default, default, cancellationToken);
-            return new SubmissionSs(result?.Submissions?[0]);
+            await _strawberryShakeClient.Submissions().MarkRetrieved(submissionId, retrieved, cancellationToken);
+            var result = await _strawberryShakeClient.Submissions().Get(submissionId, cancellationToken);
+            return new Submission
```
Comment (goatrocks) [inline IndicoV2/Submissions/SubmissionsClient.cs:150]: you should be able to use the `ToSubmissionFromSs` function below this one I believe

File: Nuget.props [hunk 2]
```diff
@@ -10,7 +10,7 @@
 		<RepositoryUrl>https://github.com/IndicoDataSolutions/indico-client-csharp</RepositoryUrl>
 		<RepositoryType>Github</RepositoryType>
 		<PackageTags>Indico</PackageTags>
-		<PackageReleaseNotes>This is the 4.x IPA platform package.</PackageReleaseNotes>
+		<PackageReleaseNotes>This is the 6.x IPA platform package.</PackageReleaseNotes>
```
Comment (goatrocks) [inline Nuget.props:13]: oh dear 😶‍🌫️ that was outta date

File: IndicoV2/Submissions/SubmissionsClient.cs (hunk 1 above)
Comment (nateshim-indico) [reply IndicoV2/Submissions/SubmissionsClient.cs]: The typing isn't working out for me due to the different return type between ListSubmissions and GetSubmission `IListSubmission_Submissions_Submissions` type versus `IGetSubmission_Submissions`

**Review by mawelborn — APPROVED** [review]
Built this and tested with a C# codebase. The call takes less than a second now!

**Pull request merged**